polaris/polaris/sep38/quote.py

```diff
@@ -0,0 +1,119 @@
+
+    return r
+
+
+quote_response_fields = [
+    "id",
+    "expires_at",
+    "price",
+    "sell_asset",
+    "sell_amount",
+    "buy_asset",
+    "buy_amount",
+]
+
+
+class QuoteAPIView(APIView):
+    parser_classes = [JSONParser]
+    renderer_classes = [JSONRenderer]
+
+    @staticmethod
+    @check_authentication()
+    def get(_: Request, quote_id: str) -> Response:
+        try:
+            quote = get_quote_by_id(quote_id)
+            return Response(model_to_dict(quote, fields=quote_response_fields))
+        except Exception as ex:
+            return render_error_response(
+                description="Error getting the quote. cause={}".format(ex.__str__()),
+                status_code=status.HTTP_400_BAD_REQUEST,
+            )
+
+    @staticmethod
+    @check_authentication()
+    def post(request: Request) -> Response:
+        from polaris.models import Quote
+
+        try:
+            r = validate_quote_request(request.data)
+        except Exception as ex:
+            return render_error_response(
+                description="Error validating the quote. cause={}".format(ex.__str__()),
+                status_code=status.HTTP_400_BAD_REQUEST,
+            )
+
+        quote = Quote()
+        quote.sell_asset = r.get("sell_asset")
+        quote.buy_asset = r.get("buy_asset")
+        quote.sell_amount = r.get("sell_amount")
+        quote.buy_amount = r.get("buy_amount")
+        quote.requested_expire_after = r.get("expire_after")
+        quote.sell_delivery_method = r.get("sell_delivery_method")
+        quote.buy_delivery_method = r.get("buy_delivery_method")
+        quote.country_code = r.get("country_code")
+
+        try:
+            exchange_pair = get_exchange_pair(
+                sell_asset=quote.buy_asset, buy_asset=quote.sell_asset
+            )
+            if exchange_pair is None:
+                return render_error_response(
+                    description="The requested quote is not provided.",
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                )
+            quote = rqi.post_quote(quote)
+            buy_decimals = get_significant_decimals(quote.buy_asset)
+            sell_decimals = get_significant_decimals(quote.sell_asset)
+            if quote.sell_amount is not None:
+                quote.buy_amount = to_decimals(
+                    float(quote.sell_amount) / float(quote.price), buy_decimals
+                )
+            else:
+                quote.sell_amount = to_decimals(
+                    float(quote.buy_amount) * float(quote.price), sell_decimals
+                )
+        except Exception as ex:
+            return render_error_response(
+                description="Error executing the quote. cause={}".format(ex.__str__()),
+                status_code=status.HTTP_400_BAD_REQUEST,
+            )
+
+        Quote.save(quote)
+        return Response(model_to_dict(quote, fields=quote_response_fields))
```

Comment:
Instead of using `model_to_dict()` and similar functions, we should write serializer classes from `rest_framework`. Take a look at our other serializer classes for reference.